core/src/test/java/org/apache/iceberg/actions/rewrite/TestBinPackStrategy.java

```diff
@@ -0,0 +1,179 @@
+  }
+
+  private RewriteStrategy defaultBinPack() {
+    return new TestBinPackStrategyImpl().options(Collections.emptyMap());
+  }
+
+  @Test
+  public void testFilteringAllValid() {
+    RewriteStrategy strategy = defaultBinPack();
+    Iterable<FileScanTask> testFiles = filesOfSize(100, 100, 100, 100, 1000);
+
+    Iterable<FileScanTask> filtered = strategy.selectFilesToRewrite(testFiles);
+
+    Assert.assertEquals("No files should be removed from the set", testFiles, filtered);
+  }
+
+  @Test
+  public void testFilteringRemoveInvalid() {
+    RewriteStrategy strategy = defaultBinPack();
+    Iterable<FileScanTask> testFiles = filesOfSize(500, 500, 500, 600, 600);
+
+    Iterable<FileScanTask> filtered = strategy.selectFilesToRewrite(testFiles);
+
+    Assert.assertEquals("All files should be removed from the set", Collections.emptyList(), filtered);
+  }
+
+  @Test
+  public void testFilteringCustomMinMaxFileSize() {
+    RewriteStrategy strategy = defaultBinPack().options(ImmutableMap.of(
+        BinPackStrategy.MAX_FILE_SIZE_BYTES, Long.toString(550 * MB),
+        BinPackStrategy.MIN_FILE_SIZE_BYTES, Long.toString(490 * MB)
+    ));
+
+    Iterable<FileScanTask> testFiles = filesOfSize(500, 500, 480, 480, 560, 520);
+    Iterable<FileScanTask> expectedFiles = filesOfSize(480, 480, 560);
+
+    Iterable<FileScanTask> filtered = strategy.selectFilesToRewrite(testFiles);
+
+    Assert.assertEquals("Should remove files that exceed or are smaller than new bounds", expectedFiles, filtered);
+  }
+
+  @Test
+  public void testGroupingMinInputFilesInvalid() {
+    RewriteStrategy strategy = defaultBinPack().options(ImmutableMap.of(
+        BinPackStrategy.MIN_NUM_INPUT_FILES, Integer.toString(5)
+    ));
+
+    Iterable<FileScanTask> testFiles = filesOfSize(1, 1, 1, 1);
+
+    Iterable<List<FileScanTask>> grouped = strategy.planFileGroups(testFiles);
+
+    Assert.assertEquals("Should plan 0 groups, not enough input files",
+        0, Iterables.size(grouped));
+  }
+
+  @Test
+  public void testGroupingMinInputFilesValid() {
+    RewriteStrategy strategy = defaultBinPack().options(ImmutableMap.of(
+        BinPackStrategy.MIN_NUM_INPUT_FILES, Integer.toString(5)
+    ));
+
+    Iterable<FileScanTask> testFiles = filesOfSize(1, 1, 1, 1, 1);
+
+    Iterable<List<FileScanTask>> grouped = strategy.planFileGroups(testFiles);
+
+    Assert.assertEquals("Should plan 1 groups since there are enough input files",
+        ImmutableList.of(testFiles), grouped);
+  }
+
+  @Test
+  public void testGroupingMinOutputFilesInvalid() {
+    RewriteStrategy strategy = defaultBinPack().options(ImmutableMap.of(
+        BinPackStrategy.MIN_NUM_OUTPUT_FILES, Integer.toString(3)
+    ));
+
+    Iterable<FileScanTask> testFiles = filesOfSize(200, 200, 200, 200, 200);
+
+    Iterable<List<FileScanTask>> grouped = strategy.planFileGroups(testFiles);
+
+    Assert.assertEquals("Should plan 1 groups since there would be 2 output files",
```

Comment:
is the error message correct?

Reply:
Nope, I copy pasted that, should be empty in this message.